lib/morse_translator.rb

```diff
@@ -40,4 +42,32 @@ def initialize
                     " " => " "}
   end
 
+  def translate(string)
+    morse_code = ""
+    string.downcase.split("").each do |letter|
+      morse_code << @dictionary[letter]
+    end
+    return morse_code
+  end
+
+  def morse_to_eng(morse_code)
+    string = ""
+    morse_code.split("  ").each do |code|
+      if code == morse_code.split("  ").last
+        string << code_to_word(code)
+      else
+        string << code_to_word(code) + " "
+      end
+    end
+    return string
+  end
+
+  def code_to_word(code)
+    word = ""
+    code.split.each do |letter|
+      word << @dictionary.key(letter)
+    end
```

Comment:
You have a lot of these `each`s that could be maps. Make sure you're working on your enumerables!